JSDoc typedef parsing fails if type and name are separated by a newline

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  3.1.0-dev.20180830

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**

 * JSDoc
 * asterisk

**Code**

```js
/** 
 * @typedef {function(string): boolean}
 * MyType
 */

/**
 * @param {MyType} func The function to call.
 * @param {string} arg The argument to call it with.
 * @returns {boolean} The return.
 */
function callIt(func, arg) {
  return func(arg);
}
```

**Expected behavior:**

No errors from `tsc --pretty`

**Actual behavior:**

```
main.js:7:12 - error TS2304: Cannot find name 'MyType'.

7  * @param {MyType} func The function to call.
             ~~~~~~
```

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

 * #26749 
 * #23667

--- a/src/compiler/parser.ts
+++ b/src/compiler/parser.ts
@@ -6856,7 +6856,7 @@ namespace ts {
 
                 function parseTypedefTag(atToken: AtToken, tagName: Identifier, indent: number): JSDocTypedefTag {
                     const typeExpression = tryParseTypeExpression();
-                    skipWhitespace();
+                    skipWhitespaceOrAsterisk();
 
                     const typedefTag = <JSDocTypedefTag>createNode(SyntaxKind.JSDocTypedefTag, atToken.pos);
                     typedefTag.atToken = atToken;

--- a/tests/baselines/reference/typedefTagWrapping.symbols
+++ b/tests/baselines/reference/typedefTagWrapping.symbols
@@ -1,0 +1,23 @@
+=== tests/cases/conformance/jsdoc/mod1.js ===
+/** 
+ * @typedef {function(string): boolean}
+ * MyType
+ */
+
+/**
+ * Tries to use a type whose name is on a different
+ * line than the typedef tag.
+ * @param {MyType} func The function to call.
+ * @param {string} arg The argument to call it with.
+ * @returns {boolean} The return.
+ */
+function callIt(func, arg) {
+>callIt : Symbol(callIt, Decl(mod1.js, 0, 0))
+>func : Symbol(func, Decl(mod1.js, 12, 16))
+>arg : Symbol(arg, Decl(mod1.js, 12, 21))
+
+  return func(arg);
+>func : Symbol(func, Decl(mod1.js, 12, 16))
+>arg : Symbol(arg, Decl(mod1.js, 12, 21))
+}
+

--- a/tests/baselines/reference/typedefTagWrapping.types
+++ b/tests/baselines/reference/typedefTagWrapping.types
@@ -1,0 +1,24 @@
+=== tests/cases/conformance/jsdoc/mod1.js ===
+/** 
+ * @typedef {function(string): boolean}
+ * MyType
+ */
+
+/**
+ * Tries to use a type whose name is on a different
+ * line than the typedef tag.
+ * @param {MyType} func The function to call.
+ * @param {string} arg The argument to call it with.
+ * @returns {boolean} The return.
+ */
+function callIt(func, arg) {
+>callIt : (func: (arg0: string) => boolean, arg: string) => boolean
+>func : (arg0: string) => boolean
+>arg : string
+
+  return func(arg);
+>func(arg) : boolean
+>func : (arg0: string) => boolean
+>arg : string
+}
+

--- a/tests/cases/conformance/jsdoc/typedefTagWrapping.ts
+++ b/tests/cases/conformance/jsdoc/typedefTagWrapping.ts
@@ -1,0 +1,20 @@
+// @noEmit: true
+// @allowJs: true
+// @checkJs: true
+// @Filename: mod1.js
+
+/** 
+ * @typedef {function(string): boolean}
+ * MyType
+ */
+
+/**
+ * Tries to use a type whose name is on a different
+ * line than the typedef tag.
+ * @param {MyType} func The function to call.
+ * @param {string} arg The argument to call it with.
+ * @returns {boolean} The return.
+ */
+function callIt(func, arg) {
+  return func(arg);
+}

